rust_build_utils/rust_utils.py

```diff
@@ -481,3 +486,75 @@ def remove_tree_or_file(path):
         shutil.rmtree(path)
     except NotADirectoryError:
         os.remove(path)
+
+
+def generate_uniffi_bindings(
+    project: Project, generator_version: str, languages: List[str], udl_path: str
+):
+    """Generate uniFFI bindings using NordSecurity/uniffi-generators docker image.
+
+    Args:
+        project (Project): Project object
+        generator_version (str): Version of the uniffi-generators docker image
+        languages (List[str]): List of languages to generate bindings for (kotlin, swift, python, cs, go)
+        udl_path (str): Path to the UDL file relative to the project root directory
+
+    Note:
+        I'm aware of "Docker SDK for Python" module. I've tried it but had potential issues on Windows and Mac
+        so it seems that calling docker via subprocess is more reliable.
+    """
+
+    class UniffiContainer:
+        def __init__(self):
+            self.container_id = None
+
+        def __enter__(self):
+            run_args = [
+                "docker",
+                "run",
+                "--rm",
+                "-d",
+                "-t",
+                "-v",
+                f"{project.get_root_dir()}:/workdir",
+                "-w",
+                "/workdir",
+            ]
+            if os.name == "posix":
+                run_args.extend(["-u", f"{os.getuid()}:{os.getgid()}"])
+            run_args.append(
+                f"ghcr.io/nordsecurity/uniffi-generators:{generator_version}"
+            )
+            self.container_id = subprocess.check_output(run_args).decode().strip()
+            return self
+
+        def __exit__(self, exc_type, exc_val, exc_tb):
+            if self.container_id:
+                subprocess.check_output(["docker", "stop", self.container_id])
```

Comment:
I assume `docker rm` is not needed, because `--rm` is used when creating the container?